src/Config.php

```diff
@@ -48,8 +48,8 @@ public function __construct($hostname, $conf = null, CacheItemPoolInterface $cac
             // If there is no cacheItemPool, use shared memory for
             // caching the configuration and channel pool.
             if (! extension_loaded('sysvshm')) {
-                throw \RuntimeException(
-                'sysvshm extension is required to use this ItemPool');
+                error_log('sysvshm extension is required to use this ItemPool');
```

Comment:
WDYT about using [trigger_error](http://php.net/manual/en/function.trigger-error.php) here instead? It will allow users to catch and handle this error however they see fit by utilizing [set_error_handler](http://php.net/manual/en/function.set-error-handler.php).